# **Tugas Day 10**

## **Import Library**

In [74]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [75]:
import pandas as pd
import numpy as np


In [76]:
df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Day 10/marketing_campaign.csv', sep='\t')
df

ID  Year_Birth   Education Marital_Status   Income  Kidhome  \
0      5524        1957  Graduation         Single  58138.0        0   
1      2174        1954  Graduation         Single  46344.0        1   
2      4141        1965  Graduation       Together  71613.0        0   
3      6182        1984  Graduation       Together  26646.0        1   
4      5324        1981         PhD        Married  58293.0        1   
...     ...         ...         ...            ...      ...      ...   
2235  10870        1967  Graduation        Married  61223.0        0   
2236   4001        1946         PhD       Together  64014.0        2   
2237   7270        1981  Graduation       Divorced  56981.0        0   
2238   8235        1956      Master       Together  69245.0        0   
2239   9405        1954         PhD        Married  52869.0        1   

      Teenhome Dt_Customer  Recency  MntWines  ...  NumWebVisitsMonth  \
0            0  04-09-2012       58       635  ...                  7   
1            1  08-03-2014       38        11  ...                  5   
2            0  21-08-2013       26       426  ...                  4   
3            0  10-02-2014       26        11  ...                  6   
4            0  19-01-2014       94       173  ...                  5   
...        ...         ...      ...       ...  ...                ...   
2235         1  13-06-2013       46       709  ...                  5   
2236         1  10-06-2014       56       406  ...                  7   
2237         0  25-01-2014       91       908  ...                  6   
2238         1  24-01-2014        8       428  ...                  3   
2239         1  15-10-2012       40        84  ...                  7   

      AcceptedCmp3  AcceptedCmp4  AcceptedCmp5  AcceptedCmp1  AcceptedCmp2  \
0                0             0             0             0             0   
1                0             0             0             0             0   
2                0             0             0             0             0   
3                0             0             0             0             0   
4                0             0             0             0             0   
...            ...           ...           ...           ...           ...   
2235             0             0             0             0             0   
2236             0             0             0             1             0   
2237             0             1             0             0             0   
2238             0             0             0             0             0   
2239             0             0             0             0             0   

      Complain  Z_CostContact  Z_Revenue  Response  
0            0              3         11         1  
1            0              3         11         0  
2            0              3         11         0  
3            0              3         11         0  
4            0              3         11         0  
...        ...            ...        ...       ...  
2235         0              3         11         0  
2236         0              3         11         0  
2237         0              3         11         0  
2238         0              3         11         0  
2239         0              3         11         1  

[2240 rows x 29 columns]

## **Soal 1 a**

In [77]:
pmf = df['NumWebVisitsMonth'].value_counts(normalize=True).sort_index() * 100
pmf

NumWebVisitsMonth
0      0.491071
1      6.830357
2      9.017857
3      9.151786
4      9.732143
5     12.544643
6     15.178571
7     17.544643
8     15.267857
9      3.705357
10     0.133929
13     0.044643
14     0.089286
17     0.044643
19     0.089286
20     0.133929
Name: proportion, dtype: float64

## **Soal 1 b**

In [78]:
# Probability of having exactly 5 web visits
prob_5_visits = pmf.get(5, 0)

# Probability of having more than 5 web visits
prob_more_than_5_visits = pmf[pmf.index > 5].sum()

prob_5_visits, prob_more_than_5_visits

(12.544642857142858, 52.232142857142854)

**Probabilitas seorang pelanggan memiliki 5 kunjungan web bulanan adalah sekitar 12.54%. Sementara itu, probabilitas seorang pelanggan memiliki lebih dari 5 kunjungan web bulanan adalah sekitar 52.23%. Jadi, pelanggan lebih mungkin memiliki lebih dari 5 kunjungan daripada tepat 5 kunjungan.**

## **Soal 2**

In [79]:
import pandas as pd
import scipy.stats as stats


# Drop rows with missing income or education values
df = df.dropna(subset=['Income', 'Education'])

# Define the two education groups to compare, e.g., "Graduation" vs "PhD"
group1 = df[df['Education'] == 'Graduation']['Income']
group2 = df[df['Education'] == 'PhD']['Income']

# Perform the two-sample t-test
t_stat, p_value = stats.ttest_ind(group1, group2, equal_var=False)

# Display the results
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpretation
alpha = 0.05
if p_value < alpha:
    print("Terdapat perbedaan signifikan dalam rata-rata pendapatan antara dua kelompok pendidikan ini.")
else:
    print("Tidak terdapat perbedaan signifikan dalam rata-rata pendapatan antara dua kelompok pendidikan ini.")


T-statistic: -2.712075660113503
P-value: 0.006780066485655928
Terdapat perbedaan signifikan dalam rata-rata pendapatan antara dua kelompok pendidikan ini.


**Dapat disimpulkan bahwa tingkat pendidikan ("Graduation" vs. "PhD") berpengaruh signifikan terhadap pendapatan. Ini mungkin menunjukkan bahwa tingkat pendidikan yang lebih tinggi (seperti PhD) terkait dengan pendapatan yang lebih tinggi atau berbeda secara signifikan dibandingkan dengan tingkat pendidikan yang lebih rendah (seperti Graduation).**